D-lama-service/Data/ProjectEntities/DataPoint.cs

```diff
@@ -10,12 +10,23 @@ public abstract class DataPoint : Entity
     {
         [Required]
         public int DataPointIndex { get; set; }
+
         [Required]
         [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
         public DateTime CreationDate { get; set; } // on create
+
         [Required]
         public DateTime UpdateDate { get; set; } // on update
+
         [Required]
         public int Version { get; set; } // increment per update
+
+        // Required foreign key property
+        public int ProjectId { get; set; }
+        // Required reference navigation to principal
+        public Project Project { get; set; } = null!;
```

Comment:
I would add the Annotation [JsonIgnore] for the Project reference i order to avoid circular references during the json serialization.
```suggestion
        [JsonIgnore]
        public Project Project { get; set; } = null!;
```

Reply:
Good point. Added your suggestion